clcache.py

```diff
@@ -765,43 +776,110 @@ def expandCommandLine(cmdline):
     return ret
 
 
+class Argument(object):
+    def __init__(self, name):
+        self.name = name
+
+    def __len__(self):
+        return len(self.name)
+
+    def __str__(self):
+        return "/" + self.name
+
+    def __eq__(self, other):
+        return self.name == other.name
+
+    def __hash__(self):
+        return hash(self.name)
+
+
+# /NAMEparameter (no space, required parameter).
+class ArgumentT1(Argument):
+    def __init__(self, name):
+        super(ArgumentT1, self).__init__(name)
```

Comment:
I think you don't need an explicit constructor here (and in the other subclasses), you can just write `pass` which will get you the `Argument` constructor.